astrapy/api.py

```diff
@@ -0,0 +1,71 @@
+import httpx
+
+from typing import cast
+
+from astrapy.types import API_RESPONSE
+from astrapy.utils import make_request
+
+
+class APIRequestError(Exception):
```

Comment:
I would add a member that contains the very httpx.RequestError, not made into a string, to avoid swallowing useful info from clients who may want to use that stuff